src/routes/productRoutes.ts

```diff
@@ -0,0 +1,21 @@
+import express from 'express';
+import {
+  createCollection,
+  createProduct,
+} from '../controllers/productController';
+import isAuthenticated from '../middleware/isAuthMiddleware';
+import checkPermission from '../middleware/checkPermissionMiddleware';
+import upload from '../middleware/multerConfig';
+
+const router = express.Router();
+
+router.post('/', isAuthenticated, checkPermission('SELLER'), createCollection);
+router.post(
+  '/:collectionId',
+  isAuthenticated,
+  checkPermission('SELLER'),
```

Comment:
You can re-check how you can leverage the use of `checkPermission` Middleware after resolving the required modification